tests/lax_numpy_test.py

```diff
@@ -3835,6 +3836,14 @@ def f(x):
 
     check_grads(f, (1.,), order=1)
 
+  def testSetPrintOptions(self):
+    x = random.normal(random.PRNGKey(0), ()) / 1000
+    y = jnp.set_printoptions(x,"unique")  
+    np.set_printoptions(floatmode="unique")
+    z = np.array(x)
+
+    self.assertEqual(y,z)
```

Comment:
It's not sensible to compare the arrays for equality. Don't you want to test a property about how the arrays print as strings? Note "exact equality with NumPy" isn't the right criterion either, but testing that particular arrays convert to particular strings might work.

Reply:
@hawkinsp you are definitely right, thank you for the suggestion. Now we have a more meaningful test that fails as expected. I think it could be further improved later though.